x-pack/solutions/security/plugins/security_solution/docs/rfcs/detection_response/customized_rule_alert_telemetry.md

```diff
@@ -0,0 +1,299 @@
+
+With this definition, we will be conservative in the fields we are defining as "functional" and only the 100%, unarguable fields will be labeled as such. Given we have a list of fields to query on in the telemetry data, alerts can be filtered upon in a more nuanced way on the telemetry side of things if needed.
+
+A full list of fields we diff on with their functional classification is located in the RFC appendix.
+
+### Schema definitions
+
+We will want to map this new field to the existing `rule_source` field mapping. Notably **NOT** as a nested field, as it would constrain us heavily for KQL queries and dashboard tooling later on down the line. 
+
+Putting it all together under one field also makes it very easy to allowlist into the alert telemetry schema they already have without including the rest of the `kibana.alert.rule.parameters` field. Including this whole parameters object could increase the size of the alert document by a lot and wouldn't really add a lot of useful data at scale. Plus, if this data *was* ever needed further down the line, it wouldn't be that difficult to add on the telemetry side of things.
+
+**Mapping example of new rule source object for prebuilt rules**
+
+```js
+rule_source: {
+    type: 'external';
+    is_customized: boolean;
+    customized_fields: Array<{
+        field_name: string;
+    }>;
+    missing_base_version: boolean;
+};
+```
+
+The mapping for our detection rules is carried over onto the alert document schema as the flattened `kibana.alert.rule.parameters` field. Changing the rule schema here should carry the same change to the detection alert schema on the kibana side of things.
+
+**Alert telemetry schema**
+
+We will want to enrich this data in the telemetry pipeline with functional field markers so that we can easily filter on this data on the telemetry side of things. We wait till the telemetry endpoint to do this comparison and enrichment because the information is not needed on the rule object or alert documents themselves - we could always calculate it on the fly if need be later on.
+
+The current implementation of the prebuilt rule alert schema uses [this list](https://github.com/elastic/kibana/blob/0ad59fab8835d67b476ffbdb82850649e70ab4d9/x-pack/solutions/security/plugins/security_solution/server/lib/telemetry/filterlists/prebuilt_rules_alerts.ts) to include alert fields onto the sanitized alert documents. In order to include the `rule_source` object and *not* the rest of the `kibana.alert.rule.parameters` field we would need to include the following mapping:
+
+```js
+'kibana.alert.rule.parameters': {
+  rule_source: true,
+}
```

Comment:
Do we really need the original `rule_source` object in the telemetry alert objects? There will be a good amount of overlap with the new `kibana.alert.rule.customizations` field we're going to add.